src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,236 @@
+                          { sbLeftAreaStart, sbLeftAreaEnd },
+                          { sbRightAreaStart, sbRightAreaEnd } };
+
+    userPrimersShowHideGroup->init(USER_PRIMERS_SHOW_HIDE_ID, tr("User primers"), wgtUserPrimers, true);
+    parametersOfPrimingSequencesShowHideGroup->init(PARAMETERS_OF_PRIMING_SEQUENCES_SHOW_HIDE_ID,
+                                                    tr("Parameters of priming sequences"),
+                                                    wgtParametersOfPrimingSequences,
+                                                    true);
+    parameters2ExcludeInWholePrimersShowHideGroup->init(PARAMETERS_2_EXCLUDE_IN_WHOLE_PRIMERS_SHOW_HIDE_ID,
+                                                        tr("Parameters to exclude in whole primers"),
+                                                        wgtParameters2ExcludeInWholePrimers,
+                                                        true);
+    areasForPrimingShowHideGroup->init(SELECT_AREAS_FOR_PRIMING_SHOW_HIDE_ID,
+                                       tr("Select areas for priming search"),
+                                       wgtAreasForPriming,
+                                       true);
+    openBackboneSequenceShowHideGroup->init(OPEN_BACKBONE_SEQUENCE_SHOW_HIDE_ID,
+                                            tr("Open the backbone sequence"),
+                                            wgtOpenBackboneSequence,
+                                            true);
+    generateSequenceShowHideGroup->init(GENERATE_SEQUENCE_SHOW_HIDE_ID,
+                                        tr("Generate a sequence"),
+                                        wgtGenerateSequence,
+                                        true);
+    otherSequencesInPCRShowHideGroup->init(OTHER_SEQUENCES_IN_PCR_REACTION_SHOW_HIDE_ID,
+                                           tr("Other sequences in PCR reaction"),
+                                           wgtOtherSequencesInPcr,
+                                           true);
+
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto seqLength = activeSequenceContext->getSequenceLength();
+
+    sbLeftAreaStart->setValue(1);
+    sbLeftAreaEnd->setValue(seqLength / 10);
+    sbRightAreaStart->setValue(seqLength / 10 * 5);
+    sbRightAreaEnd->setValue(seqLength / 10 * 6);
+
+    connect(pbStart, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_start);
+    connect(tbLeftAreaSelectManually, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually);
+    connect(tbRightAreaSelectManually, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually);
+
+    U2WidgetStateStorage::restoreWidgetState(savableWidget);
+
+    sl_updateMaxValues();
+    for (auto minSb : minMaxValuesSb.keys()) {
```

Comment:
qAsConst

Reply:
error C2280: 'void qAsConst<QList<Key>>(const T &&)': attempting to reference a deleted function

Reply:
This issue was not fixed.

Reply:
Fixed